homeassistant/components/wmspro/scene.py

```diff
@@ -0,0 +1,58 @@
+"""Support for scenes provided by WMS WebControl pro."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from wmspro.scene import Scene as WMS_Scene
+
+from homeassistant.components.scene import Scene
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WebControlProConfigEntry
+from .const import ATTRIBUTION, DOMAIN, MANUFACTURER
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WebControlProConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the WMS based scenes from a config entry."""
+    hub = config_entry.runtime_data
+
+    async_add_entities(
+        WebControlProScene(config_entry.entry_id, scene)
+        for scene in hub.scenes.values()
+    )
+
+
+class WebControlProScene(Scene):
+    """Representation of a WMS based scene."""
+
+    _attr_attribution = ATTRIBUTION
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, config_entry_id: str, scene: WMS_Scene) -> None:
```

Comment:
Personal preference, have the bigger objects first

Reply:
We had this exchange before regarding the base entity class and destinations. Since I cannot base the Scenes on the base entity class (but other device types will follow doing that, don't worry), I would like to keep the parameter order at least aligned with that:
https://github.com/home-assistant/core/blob/67f67a02f87f108c9bf65598c49b911065ec9e65/homeassistant/components/wmspro/entity.py#L20